R/mpf_exporters.R

```diff
@@ -463,3 +461,175 @@ export_as_rtf <- function(x,
     restxt
   }
 }
+
+
+#' Export as PDF
+#'
+#' The PDF output is based on the ASCII output created with [toString()]
+#'
+#' @inheritParams export_as_txt
+#' @param file file to write, must have `.pdf` extension
+#' @param width Deprecated, please use `pg_width` or specify
+#'   `page_type`. The width of the graphics region in inches
+#' @param height Deprecated, please use `pg_height` or specify
+#'   `page_type`. The height of the graphics region in inches
+#' @param fontsize Deprecated, please use `font_size`. The size of
+#'   text (in points)
+#' @param margins numeric(4). The number of lines/characters of margin on the
+#'   bottom, left, top, and right sides of the page.
+#'
+#' @importFrom grDevices pdf
+#' @importFrom grid textGrob grid.newpage gpar pushViewport plotViewport unit grid.draw
+#'   convertWidth convertHeight grobHeight grobWidth
+#'
+#' @details By default, pagination is performed with default
+#' `cpp` and `lpp` defined by specified page dimensions and margins.
+#' User-specified `lpp` and `cpp` values override this, and should
+#' be used with caution.
+#'
+#' Title and footer materials are also word-wrapped by default
+#' (unlike when printed to the terminal), with `cpp`, as
+#' defined above, as the default `max_width`.
+#'
+#' @seealso [export_as_txt()]
+#'
+#' @importFrom grid textGrob get.gpar
+#' @importFrom grDevices dev.off
+#' @importFrom tools file_ext
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' tf <- tempfile(fileext = ".pdf")
+#' export_as_pdf(basic_matrix_form(mtcars), pg_width = 4, file = tf, pg_height = 4)
```

Comment:
First example is failing:

```
Error in `mf_col_widths<-`(`*tmp*`, value = widths) : 
  Number of column widths (12) does not match number of columns in strings matrix (2).
```

Can you update the parameters so that it works?